jwst/datamodels/schemas/dark.schema.yaml

```diff
@@ -1,25 +1,30 @@
 title: Dark current data model
 allOf:
 - $ref: referencefile.schema.yaml
+- $ref: instrument_name.schema.yaml
+- $ref: detector.schema.yaml
+- $ref: filter_core.schema.yaml
+- $ref: exposure_type.schema.yaml
+- $ref: subarray.schema.yaml
 - type: object
   properties:
     data:
       title: Dark current array
       fits_hdu: SCI
       default: 0.0
-      ndim: 3
+      ndim: 4
       datatype: float32
     dq:
       title: 2-D data quality array for all planes
       fits_hdu: DQ
       default: 0
-      ndim: 2
-      datatype: uint32
+      ndim: 4
+      datatype: int16
     err:
       title: Error array
       fits_hdu: ERR
       default: 0.0
-      ndim: 3
+      ndim: 4
```

Comment:
The darks for all near-IR instruments use 3D SCI and ERR arrays, and a 2D DQ array, as was listed in the previous version of the dark.schema. It's only MIRI darks, which have their own dedicated dark model, that use 4D arrays.